### ISSUE TITLE
help-wanted - Shared pages across TOCs loses context

### ISSUE THREAD

**Issue by erikcvisser** [dotnet/docfx#5606]
Hi there, first of all: thanks for this great tool!
I have started creating documentation for a project of mine and encountered a use case that I wasn't able to solve. Hopefully someone can help me out or confirm that it is not possible currently.

**Operation System**: (`Windows` or `Linux` or `MacOS`)
_all_
**DocFX Version Used**: 
2.49

**Template used**: (`default` or `statictoc` or contain custom template)
default / custom

**Steps to Reproduce**:

1. Create following files:
  /section 1/toc.yml
  /section 2/toc.yml
  /shared/logging-in.md
  /shared/toc.yml
2. reference the shared items in both section1 and section1
```
  - name: Logging in  
    href: ../shared/toc.yml
```
3. Navigate to section 2, and click on the link 'Loging in'

**Expected Behavior**:
I would expect to open the 'logging in' page, but keep the context (breadcrumbs, side menu) from section 2

**Actual Behavior**:
It does open the 'logging in' page, but instead switches context (breadcrumbs, side menu) to the first subfolder where the shared/toc.yml is mentioned, in this case section 1.

**Comment by KalleOlaviNiemitalo** [comment]
If you can tolerate having a different URL for each context, then I think you can use [File Inclusion](https://dotnet.github.io/docfx/spec/docfx_flavored_markdown.html#file-inclusion), or have your build system copy the Markdown file before it runs DocFX.

If you must have the same URL in both contexts, then it can perhaps be done by making the TOC-populating JavaScript code maintain the context information in [Window.sessionStorage](https://developer.mozilla.org/docs/Web/API/Window/sessionStorage), which [DocFX already uses for the TOC filter](https://github.com/dotnet/docfx/blob/d8f389f391d31dcc0b943fb5bf58077441a23dbf/src/docfx.website.themes/default/styles/docfx.js#L448-L451). However, the `statictoc` template avoids using client-side JavaScript for the TOC, so the feature would be incompatible with that.

**Comment by erikcvisser** [comment]
@KalleOlaviNiemitalo thank you for your response and apologies for the late testing.

I indeed want to preserve the URL of the context, so a different one for /section-1/ and /section-2/.
I have now successfully achieved what I wanted using File Inclusion. This solution does require that every /section-X/ folder contains the exact same files as /shared/. It may not be the most neat solution, but gets the job done.

Thanks again!

For future reference: in /section-1/ I now have the same MarkDown files as in /shared/, but with the following content:

`[!include[<log-in>](../../shared/logging-in.md)]`
